annotation_tool.py

```diff
@@ -0,0 +1,245 @@
+import datetime
+import json
+import os
+
+import fiona
+import ipywidgets
+import leafmap
+import solara
+
+from leafmap import toolbar
+
+#
+# Application variables
+#
+# Map variables
+center = solara.reactive((20, 0))
+zoom = solara.reactive(4)
+
+# STAC creation variables
+stac_annotation_filename = solara.reactive("data.geojson")
+stac_asset_link = solara.reactive("")
+stac_asset_title = solara.reactive("")
+stac_item_link = solara.reactive("")
+stac_collection_id = solara.reactive("")
+stac_file_name = solara.reactive("")
+stac_item_id = solara.reactive("")
+
+# Behavior variables
+continuous_update = solara.reactive(True)
+show_instruction_message = solara.reactive(True)
+
+
+#
+# Utility functions
+#
+def get_annotation_bbox(filename=stac_annotation_filename.value):
+    """
+    Create bounding box for a given annotation file
+
+    Args:
+        filename: File name of the geojson annotation file
+
+    Returns: Bounding box in list form
+
+    """
+    annotations = fiona.open(filename)
+    return list(annotations.bounds)
+
+
+def create_stac_item(
+    filename=stac_annotation_filename.value,
+    stac_id=stac_item_id.value,
+    stac_title=stac_asset_title.value,
+    stac_asset=stac_asset_link.value,
+    collection_id=stac_collection_id.value,
+    item_link=stac_item_link.value,
+):
+    """
+    Create a STAC item stub from a geojson annotation file, which
+    can later be expanded.
+
+    Args:
+        filename: Geojson annotation file name/path
+        stac_id: ID for new STAC item
+        stac_asset: Link to the asset, which is the hosted Geojson annotation file
+        stac_title: Title for the Geojson annotation file asset
+        collection_id: STAC collection ID, if applicable
+        item_link: STAC asset link of EO data used to create annotation
+
+    """
+    bbox = get_annotation_bbox(filename)
+    file_size = os.stat(filename).st_size
+    item = {
+        "stac_version": "1.0.0",
+        "type": "Feature",
+        "id": stac_id,
+        "bbox": bbox,
+        "geometry": {"type": "Polygon", "coordinates": bbox},
+        "properties": {"created": datetime.datetime.now().isoformat()},
```

Comment:
If the set of classes is known and that the annotation is `raster`-typed, this extension could apply as well: 
https://github.com/stac-extensions/classification
Can include details such as `color_hint`, `nodata: true|false` and class `name`/`value`. 

Otherwise, `label` is sufficient (and preferable for Features).